pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/DistributedTaskLockManager.java

```diff
@@ -0,0 +1,423 @@
+            }
+          }
+        }
+
+        LOGGER.info("Found {} lockNodePaths for controller instance: {}, list: {}, first lockNodePath cached: {}",
+            allLockNodePathsForController.size(), _controllerInstanceId, allLockNodePathsForController, lockNodePath);
+
+        if (lockNodePath != null && allLockNodePathsForController.size() == 1) {
+          // Check if we got the lowest sequence number (i.e., we're first in line)
+          List<String> allChildren = _propertyStore.getChildNames(lockBasePath, AccessOption.PERSISTENT);
+          if (allChildren != null && !allChildren.isEmpty()) {
+            allChildren.sort(TASK_LOCK_SEQUENCE_ID_COMPARATOR); // Sort by sequence number
+            String ourNode = lockNodePath.substring(lockNodePath.lastIndexOf('/') + 1);
+            if (ourNode.equals(allChildren.get(0))) {
+              // We have the lock!
+              LOGGER.info("Acquired lock with ephemeral node: {}", lockNodePath);
+              return new TaskLock(lockKey, _controllerInstanceId, currentTimeMs, lockNodePath);
+            } else {
+              // Someone else has the lock, clean up our node
+              boolean status = _propertyStore.remove(lockNodePath, AccessOption.EPHEMERAL);
+              LOGGER.info("Did not get lock, removing ephemeral node: {}, return status: {}", lockNodePath, status);
+              return null;
+            }
+          } else {
+            // No children found, something went wrong, clean up
+            boolean status = _propertyStore.remove(lockNodePath, AccessOption.EPHEMERAL);
+            LOGGER.warn("No children found under {}. Remove lockNodePath status: {} for node: {}. Something must have "
+                    + "gone wrong", lockBasePath, status, lockNodePath);
+            return null;
+          }
+        } else {
+          // Could not find our node path, or found too many paths for the same controller, cleanup failed creation
+          LOGGER.warn("Either lockNodePath: {} wasn't found, or too many locks ({}) found for the same controller: {},"
+                  + "list of locks: {}", lockNodePath, allLockNodePathsForController.size(), _controllerInstanceId,
+              allLockNodePathsForController);
+
+          if (lockNodePath != null) {
+            boolean status = _propertyStore.remove(lockNodePath, AccessOption.EPHEMERAL);
+            LOGGER.warn("Remove lockNodePath status: {} for path: {}", status, lockNodePath);
+          }
+          return null;
+        }
+      }
+      return null;
+    } catch (Exception e) {
+      LOGGER.error("Error creating ephemeral lock under path: {}, lockKey: {}", lockBasePath, lockKey, e);
+      return null;
+    }
+  }
+
+  private void ensureBasePaths() {
+    try {
+      // Ensure minion task metadata base path exists
+      String basePath = getBasePath();
+      if (!_propertyStore.exists(basePath, AccessOption.PERSISTENT)) {
+        ZNRecord baseRecord = new ZNRecord("MINION_TASK_METADATA");
```

Comment:
the basepath should already be created in some other code flow right ? I am not able to find it, so not sure if this is true. but given that the path exists in ZK, it should be created somewhere. 

and if it is already created somewhere else, should we call that code path here rather than having duplicate code to create the path with required znode metadata / configs like AccessOption.PERSISTENT and baseRecord, etc ?

Reply:
removed the base path creation, parent ZNodes are anyways created and I was able to test using quickstart that this ensureBasePaths is not needed